concent_api/core/models.py

```diff
@@ -73,3 +75,216 @@ class MessageAuth(Model):
+    subtask_results_accepted    = models.ForeignKey(StoredMessage, null = True)
+    subtask_results_rejected    = models.ForeignKey(StoredMessage, null = True)
+
+
+class ReceiveQueue(models.Model):
+    """
+    Stores information about messages to be returned from the `receive` or `receive-out-of-band` endpoint.
+    """
+
+    class ConcentResponseType(ChoiceEnum):
+        ForceReportComputedTask         = 'ForceReportComputedTask'
+        ForceReportComputedTaskResponse = 'ForceReportComputedTaskResponse'
+        VerdictReportComputedTask       = 'VerdictReportComputedTask'
+        ForceGetTaskResultRejected      = 'ForceGetTaskResultRejected'
+        ForceGetTaskResultFailed        = 'ForceGetTaskResultFailed'
+        ForceGetTaskResultUpload        = 'ForceGetTaskResultUpload'
+        ForgetGetTaskResultDownload     = 'ForgetGetTaskResultDownload'
+        ForceSubtaskResults             = 'ForceSubtaskResults'
+        SubtaskResultsSettled           = 'SubtaskResultsSettled'
+        ForceSubtaskResultsResponse     = 'ForceSubtaskResultsResponse'
+        SubtaskResultsRejected          = 'SubtaskResultsRejected'
+
+    class ReceiveQueueType(ChoiceEnum):
+        Receive             = 'receive'
+        ReceiveOutOfBand    = 'receive_out_of_band'
+
+    response_type        = models.CharField(max_length = 32, choices = ConcentResponseType.choices())
+    client               = models.ForeignKey(Client)
+    queue                = models.CharField(max_length = 32, choices = ReceiveQueueType.choices())
+
+    # TRUE if the client has already fetched the message.
+    delivered            = models.BooleanField(default = False)
+
+    subtask              = models.ForeignKey(Subtask)
+    created_at           = models.DateTimeField(default = get_current_utc_timestamp)
+
+
+class AbstractStoredMessage(models.Model):
+
+    timestamp = models.DateTimeField()
+    signature = Base64Field(max_length = 64)
+
+    class Meta:
+        abstract = True
+
+
+class StoredTaskToCompute(AbstractStoredMessage):
+
+    requestor_id                    = models.CharField(max_length = 32)
+    requestor_public_key            = Base64Field(max_length = 64)
+    requestor_ethereum_public_key   = Base64Field(max_length = 64)
+    provider_id                     = models.CharField(max_length = 32)
+    provider_public_key             = Base64Field(max_length = 64)
+    provider_ethereum_public_key    = Base64Field(max_length = 64)
+    compute_task_def                = JSONField()
+    package_hash                    = models.CharField(max_length = 32)
+    concent_enabled                 = models.BooleanField(default = False)
+
+
+class StoredReportComputedTask(AbstractStoredMessage):
+
+    subtask_id          = models.CharField(max_length = 32)
+    result_type         = models.PositiveSmallIntegerField()
+    computation_time    = models.IntegerField()
+    node_name           = models.CharField(max_length = 32)
```

Comment:
Where did you get those lengths from?

1) Please create a constant for each of them
2) golem-messages should have validations that don't allow creating messages with values longer than that. Otherwise we can run into a situation where we receive a message that can't be stored in our database.

This applies to all string fields, not only this one.